source/common/formatter/http_specific_formatter.cc

```diff
@@ -374,6 +375,13 @@ HttpBuiltInCommandParser::getKnownFormatters() {
          [](const std::string&, absl::optional<size_t>&) {
            return std::make_unique<AccessLogTypeFormatter>();
          }}},
+       {"ACCESS_LOG_ID",
+        {CommandSyntaxChecker::COMMAND_ONLY,
+         [](const std::string&, absl::optional<size_t>&) {
+           static thread_local Random::RandomGeneratorImpl random_generator;
+           const std::string uuid = random_generator.uuid();
+           return std::make_unique<PlainStringFormatter>(uuid);
+         }}},
```

Comment:
This seems a good feature. But I think you could move this command to the `getKnownStreamInfoFormatterProviders` because typically this command doesn't depend on the http context.

And `UNIQUE_ID` would be better than `ACCESS_LOG_ID` because some users may wants to use this command to generate UUID in the header mutation or other scenarios. We needn't to limit this command must be id of ACCESS_LOG.

And the `RandomGeneratorImpl` self is thread-safe, the `thread_local` is unnecessary.

And the current implementation means the same uuid will be returned except we refreshed the configuration.

/wait

Reply:
Thanks for the review and sharing the feedback @wbpcode - I will work on the changes. 

On the note of current implementation generates the same uuid except when configuration is refreshed - It must be creating new uuid everytime the parser processes the substitution string %ACCESS_LOG_ID% in this case. The UT covers that. Can you please clarify if I am missing anything on this note?

Reply:
See, the command will be parsed only once for same configuration (like same listener HCM). And the outputted formatter will be used repeatedly by all requests until the whole configuration is refreshed by the LDS. 

You should validate the output from same formatter is unique in the UT rather than to compare two outputs from different formatters.

Reply:
@wbpcode Addressed all the concerns from the feedback